### ISSUE TITLE
Module apt_repository doesn't let you set owner/group/permissions

### ISSUE THREAD

**Issue by jjshoe** [ansible/ansible#6908]
##### Issue Type:

Security Report
##### Ansible Version:

ansible 1.5.4
##### Environment:

OSX -> Ubuntu, versions don't matter
##### Summary:

When using apt_repository module you can't specify an owner, a group, or permissions. You also can't specify a file name. The file name is dynamically created off of the contents of the file so you can't easily adjust these values after the fact.

If you have an apt repository with credentials you want to connect, like using AWS's s3, anyone can open the file since it's created with mode 0644. 

I would recommend adding an option to both name the file, and set owner, group, and mode. 
##### Steps To Reproduce:

```
- name: apt | add the repository
  sudo: yes
  apt_repository: repo="deb s3://{{ apt_user }}:{{ apt_password }}@s3.amazonaws.com/my-repo/ stable main" 
```
##### Expected Results:

The ability to specify user, group, owner, and file name so safety of credentials can be guaranteed.
##### Actual Results:

A file I can't manage due to a dynamically generated file name.

**Comment by jjshoe** [comment]
So what I think should happen to this module:

1) Make use of the common file attributes stuff so owner, group, and mode can be specified.
2) No dest should need to be specified, but the logic in naming the file should change:

2.1) Check for anisble_[md5 of the repo option].list
2.2) Check for the fubar's current implementation of the file name
2.2.1) Make changes if requested
2.2.3) Rename the file to anisble_[md5 of the repo option].list

This is the best possible option I can think of, to maintain current behavior, however you still end up fubar'd if you change the repo line, leaving behind an old *.list. I guess supporting dest still needs to be debated.

**Comment by mpdehaan** [comment]
Hi @jjshoe, we obviously take security reports seriously -- in the future, should you wish to report a security issue, you should disclose directly to security@ansible.com

I agree that I don't like the name generation that is going on here and like your "2)" series suggestions for upgrades that won't break existing users.

As for the permissions, we'll investigate whether we can apply a better default -- it would be nice if the user did not have to supply additional options.

**Comment by jjshoe** [comment]
The only thing that sucks by not specifying a name is you can't manage a file with multiple repo's specified, like you typically find: https://gist.github.com/jjshoe/6f994597f3684dbca991

I believe name should be optional, but should be the key if specified to enable you to specify with_items easily and handle removing all existing repos, and replacing with entirely new ones.

**Comment by jimi-c** [comment]
Closing this, as the above commit should fix this issue. If you continue seeing any problems related to this, or if you have any further questions related to this issue, please let us know. Thanks!

**Comment by jjshoe** [comment]
Uh, so, this PR would have fixed it: https://github.com/jjshoe/ansible/commit/b81c73cfced216e598f2148d2bcad776b3e68b38

Things to note about @jctanner 's PR and mine: NEITHER addresses existing repo's on disk. This means everyone who uses a repo with a u/p will get a change, when they shouldn't, the next time they run their playbook. It also means a repo is left around whose name contains the credentials still. 

@jctanner - I think you might want to whack this too: https://github.com/ansible/ansible/commit/db36c41c3bfaf4e147a73740a8507a2a2c152246#diff-04742c006c09cebb9fefd66404971a68R371

**Comment by mpdehaan** [comment]
I agree this shouldn't be closed yet as it should also take care of the permissions aspect.

**Comment by mpdehaan** [comment]
email to ansible-announce and ansible-project have been sent, so I'm closing this out!  Thanks all!

This will be included in 1.6.
